### ISSUE TITLE
Stream #0: not enough frames to estimate rate; consider increasing probesize

### ISSUE THREAD

**Issue by thoughtgap** [danielquinn/paperless#280]
After installing paperless and the dependencies on a Raspberry Pi running Raspbian Strech everything works fine, except the documents don't get rotated. It seems that unpaper is not doing its job. The consumer output looks like this for all the documents:

```
Starting document consumer at /home/pi/paperless/in
Parsers available: RasterisedDocumentParser
Consuming /home/pi/paperless/in/20170719Z - Correspondent - Title.pdf
Processing sheet #1: /tmp/paperless/paperlessl0qcue77/convert-0002.pnm -> /tmp/paperless/paperlessl0qcue77/convert-0002.unpaper.pnm
[pgm_pipe @ 0x12eacd0] Stream #0: not enough frames to estimate rate; consider increasing probesize
[image2 @ 0x12ed610] Using AVStream.codec to pass codec parameters to muxers is deprecated, use AVStream.codecpar instead.
[image2 @ 0x12ed610] Encoder did not produce proper pts, making some up.
Processing sheet #1: /tmp/paperless/paperlessl0qcue77/convert-0001.pnm -> /tmp/paperless/paperlessl0qcue77/convert-0001.unpaper.pnm
[pgm_pipe @ 0x8d9cd0] Stream #0: not enough frames to estimate rate; consider increasing probesize
out of deviation range - NO ROTATING
[image2 @ 0x8dc5d0] Using AVStream.codec to pass codec parameters to muxers is deprecated, use AVStream.codecpar instead.
[image2 @ 0x8dc5d0] Encoder did not produce proper pts, making some up.
Processing sheet #1: /tmp/paperless/paperlessl0qcue77/convert-0000.pnm -> /tmp/paperless/paperlessl0qcue77/convert-0000.unpaper.pnm
[pgm_pipe @ 0x10f6cd0] Stream #0: not enough frames to estimate rate; consider increasing probesize
[image2 @ 0x10f9610] Using AVStream.codec to pass codec parameters to muxers is deprecated, use AVStream.codecpar instead.
[image2 @ 0x10f9610] Encoder did not produce proper pts, making some up.
OCRing the document
Parsing for deu
Parsing for deu
Parsing for deu
Completed
Document 20170719000000: Title consumption finished
```

The config Parameters do not seem to influence this error.

Searching the web for the error produces [various](https://www.bountysource.com/issues/26485315-unpaper-message-image2-0x9a2d80-encoder-did-not-produce-proper-pts-making-some-up) [mentions](https://debianforum.de/forum/viewtopic.php?t=163155) but no solution. 

Did anyone get this running on a Raspberry Pi?

Thanks!

**Comment by danielquinn** [comment]
I wonder if this is a Pi thing or an issue for all platforms?  I can't remember the last time I tested Paperless with landscape documents.  I've tagged this as `question` and will (eventually) take a look at my own instance's output to see if I get the same message.  For all I know, unpaper isn't being told to do the rotation, and this message is just a common warning.

**Comment by thoughtgap** [comment]
Sorry for the confusion - my documents are not in landscape format, it's just that my scanner   scans the documents slightly askew and I kind of assumed unpaper would fix this for me. From the [unpaper README](https://github.com/Flameeyes/unpaper):

> The program also tries to detect misaligned centering and rotation of pages and will automatically straighten each page by rotating it to the correct angle. This process is called "deskewing".

As my documents arrive in paperless with the the same rotation and these errors were thrown I supposed both issues were connected.

**Comment by danielquinn** [comment]
Ah ok, you're missing one key bit of info: Paperless doesn't modify your documents.  Instead, it makes temporary copies and attempts to optimise them for OCR then deletes the temporary copies.  The process is:

1. Make a copy of your PDF as a series rasterised greyscale images
2. Apply unpaper to each image individually
3. OCR each image
4. Save the text to the database
5. Delete all of the above temporary files
6. Encrypt & store your original.

So if you're not seeing any changes to your PDF, that's as-expected.

**Comment by thoughtgap** [comment]
Got it! Thanks for the clarification. I will leave this issue open for the comparison of the logs.

**Comment by apastuszak** [comment]
I just set up tesseract "bare metal" and am getting this same warning.  Is there some way I can increase the probesize, or is this out of my control?

**Comment by danielquinn** [comment]
I just did some more digging on this and it appears to be an [issue with unpaper](https://github.com/Flameeyes/unpaper/pull/42#issuecomment-271548605).  Apparently it's taking to a shared API and that's where the problem is.

So this isn't really something I can fix in Paperless unfortunately.  Unpaper isn't killing the consumption process though, right?  It's just throwing out this warning?

**Comment by apastuszak** [comment]
yes, it's just throwing out a warning.  So, I'll leave it alone for now or
file a bug with unpaper.
